tools/src/npm/README.md

```diff
@@ -0,0 +1,147 @@
+# TODO
+
+Reduce file interactions:
+https://docs.npmjs.com/cli/v8/commands/npm-install#package-lock
+`--no-package-lock`
+
+Reduce target install folder
+https://docs.npmjs.com/cli/v8/commands/npm-install#global (explains the prefix flag)
+`--prefix=${join('tmp',packageName,'node_modules')}`
+
+# NPM Nx Plugin
+
+A Nx plugin that adds targets that help to work with packages published to NPM.
+
+## Usage
+
+Register the plugin in your `nx.json`
+
+```jsonc
+// nx.json
+{
+  //...
+  "plugins": ["tools/npm/npm.plugin.ts"]
+}
+```
+
+### Options
+
+You can configure the plugin by providing options object in addition to the plugin path
+
+**Options:**
+
+| Name                 | Type                   | Default                     | Description                                     |
+| -------------------- | ---------------------- | --------------------------- | ----------------------------------------------- |
+| `verbose`            | `boolean`              | `false`                     | Log additional information.                     |
+| `tsconfig`           | `string`               | `tools/tsconfig.tools.json` | The tsconfig file to use.                       |
+| `npmCheckScript`     | `string`               | `check-package-range.ts`    | The script to execute when checking a package.  |
+| `publishableTargets` | `string` or `string[]` | `["publishable"]`           | The targets that mark a project as publishable. |
+
+Example:
+
+```jsonc
+// nx.json
+{
+  //...
+  "plugins": [
+    {
+      "plugin": "tools/npm/npm.plugin.ts",
+      "options": {
+        "tsconfig": "tools/tsconfig.tools.json",
+        "npmCheckScript": "check-package-range.ts",
+        "publishableTargets": ["add-to-npm-registry"]
+      }
+    }
+  ]
+}
+```
+
+### Targets
+
+> [!NOTE]
+> A project can be marked as publishable by owning target included in `project.json`
+> Default is`publishable`.
```

Comment:
These docs need to be updated now that tags are used instead of targets.

```suggestion
### Nx tags

> [!NOTE]
> A project can be marked as publishable using `tags` in `project.json`.
> Default tag name is `publishable`.
```